src/game/common/system/archivefile.cpp

```diff
@@ -82,48 +82,49 @@ void ArchiveFile::Attach_File(File *file)
     m_backingFile = file;
 }
 
-void ArchiveFile::Get_File_List_From_Dir(Utf8String const &subdir,
+void ArchiveFile::Get_File_List_In_Directory(Utf8String const &subdir,
     Utf8String const &dirpath,
     Utf8String const &filter,
     std::set<Utf8String, rts::less_than_nocase<Utf8String>> &filelist,
     bool search_subdir) const
 {
     Utf8String path = dirpath;
+    path.To_Lower();
     Utf8String token;
-    DetailedArchiveDirectoryInfo const *dirp = &m_archiveInfo;
+    DetailedArchivedDirectoryInfo const *dirp = &m_archiveInfo;
+    path.Next_Token(&token, "\\/");
 
-    // Lower case for matching and get first item of the path.
-    path.To_Lower();
+    while (token.Get_Length() > 0) {
+        auto it = dirp->directories.find(token);
 
-    // Go to the last InfoNode in the path to extract file contents from.
-    for (path.Next_Token(&token, "\\/"); token.Is_Not_Empty(); path.Next_Token(&token, "\\/")) {
-        // If an element of the path doesn't have a node for our next directory, return.
-        if (dirp->directories.find(token) == dirp->directories.end()) {
+        if (it == dirp->directories.end()) {
```

Comment:
hm seems to invoke op!= here in original...

Reply:
fixed.